### PR TITLE
feat: dynamischer fzf-Header-Umbruch (header-wrap)

### LINKED ISSUE #302: feat: fzf-Header bei schmalen Fenstern umbrechen statt abschneiden
## Zusammenfassung

fzf-Header werden bei schmalen Terminalen/Fenstern mit `··` (Ellipsis) abgeschnitten. Betroffene Keybinding-Hinweise sind dann nicht sichtbar — genau die Information, die der User braucht, verschwindet.

**Lösung:** Mehrzeilige Header (statisch mit `$'\n'` oder dynamisch via `resize`-Event + `transform-header`) einsetzen, um bei schmalen Fenstern alle Keybindings sichtbar zu halten.

## Motivation

- **24 fzf-Header** im Repository → 8 davon haben >50 Zeichen
- Bei Fenstern unter ~60 Spalten wird der Header abgeschnitten
- Besonders problematisch: `gh-pr` (76 Zeichen) ist schon bei normalen Split-Layouts zu lang
- Benutzer sieht nicht, welche Keybindings verfügbar sind → schlechte UX

### Alle 24 Header sortiert nach Zeichenanzahl

| # | Zeichen | Funktion | Datei | Header-Text |
|---|---------|----------|-------|-------------|
| 1 | **76** | `gh-pr()` | `gh.alias:43` | `Enter: Checkout/Browser \| Ctrl+D: Diff \| Ctrl+O: Browser \| Ctrl+S: Open/Alle` |
| 2 | **65** | `cmds()` | `fzf.alias:137` | `Enter: Übernehmen \| Ctrl+S: tldr ↔ Code \| Ctrl+E: Datei editieren` |
| 3 | **65** | `zj()` | `zoxide.alias:49` | `Enter: Wechseln \| Ctrl+D: Eintrag löschen \| Ctrl+Y: Pfad kopieren` |
| 4 | **58** | `brew-add()` | `brew.alias:248` | `[F]=Formula [C]=Cask \| Enter: Installieren \| Tab: Mehrfach` |
| 5 | **55** | `gh-issue()` | `gh.alias:69` | `Enter: Browser \| Ctrl+E: Bearbeiten \| Ctrl+S: Open/Alle` |
| 6 | **55** | `brew-rm()` | `brew.alias:272` | `[F]=Formula [C]=Cask \| Enter: Entfernen \| Tab: Mehrfach` |
| 7 | **55** | `dotedit()` | `dotfiles.alias:90` | `Enter: Editieren \| Ctrl+Y: Pfad kopieren \| Esc: Beenden` |
| 8 | **54** | `gh-gist()` | `gh.alias:134` | `Enter: Anzeigen \| Ctrl+E: Bearbeiten \| Ctrl+O: Browser` |
| 9 | **50** | `procs()` | `fzf.alias:52` | `Enter: Beenden \| Tab: Mehrfach \| Ctrl+S: Apps/Alle` |
| 10 | **45** | `gh-run()` | `gh.alias:105` | `Enter: Logs \| Ctrl+R: Rerun \| Ctrl+O: Browser` |
| 11 | **44** | `vars()` | `fzf.alias:186` | `Enter: Export → Edit \| Ctrl+Y: Wert kopieren` |
| 12 | **43** | `rg-live()` | `rg.alias:80` | `Enter: Datei öffnen \| Ctrl+Y: Pfad kopieren` |
| 13 | **42** | `git-stage()` | `git.alias:104` | `Enter: Add \| Tab: Mehrfach \| Ctrl+R: Reset` |
| 14 | **41** | `git-stash()` | `git.alias:124` | `Enter: Apply \| Ctrl+P: Pop \| Ctrl+D: Drop` |
| 15 | **40** | `bat-theme()` | `bat.alias:45` | `Enter: Theme aktivieren \| Esc: Abbrechen` |
| 16 | **39** | `jump()` | `fd.alias:64` | `Enter: Wechseln \| Ctrl+Y: Pfad kopieren` |
| 17 | **38** | `git-log()` | `git.alias:66` | `Enter: Anzeigen \| Ctrl+Y: SHA kopieren` |
| 18 | **38** | `7zf()` | `7z.alias:49` | `Enter=Entpacken │ Ctrl+Y=Pfad kopieren` |
| 19 | **37** | `pick()` | `fd.alias:76` | `Enter: Öffnen \| Ctrl+Y: Pfad kopieren` |
| 20 | **34** | `help()` | `fzf.alias:83` | `Enter: öffnen \| Ctrl+S: man ↔ tldr` |
| 21 | **33** | `git-branch()` | `git.alias:87` | `Enter: Checkout \| Ctrl+D: Löschen` |
| 22 | **31** | `gh-repo()` | `gh.alias:120` | `Enter: Klonen \| Ctrl+O: Browser` |
| 23 | **30** | `procs()` Linux | `fzf.alias:59` | `Enter: Beenden \| Tab: Mehrfach` |
| 24 | **16** | History (Widget) | `init.zsh:36` | `Ctrl+Y: Kopieren` |

### Risiko-Klassen nach Fensterbreite

| Risikoklasse | Zeichen | Betroffene Funktionen | Abgeschnitten ab |
|--------------|---------|----------------------|------------------|
| 🔴 Hoch | >60 | `gh-pr`, `cmds`, `zj` | ~65 Spalten |
| 🟡 Mittel | 50–60 | `brew-add`, `gh-issue`, `brew-rm`, `dotedit`, `gh-gist`, `procs` | ~55 Spalten |
| 🟢 Niedrig | <50 | Alle anderen | ~50 Spalten |

### Nebenbefund: Inkonsistenter Separator in `7zf()`

`7zf()` verwendet abweichend:
- `=` statt `:` als Key-Value-Separator (`Enter=Entpacken` statt `Enter: Entpacken`)
- `│` (Box Drawing U+2502) statt `|` (Pipe U+007C) als Aktions-Separator

Alle anderen 22 Header nutzen einheitlich `: ` und `|`.

### Ansatz A: Statischer Zweizeiler (einfach)

Header bei langen Texten mit `$'\n'` aufteilen:

```zsh
--header=$'Enter: Checkout/Browser | Ctrl+D: Diff\nCtrl+O: Browser | Ctrl+S: Open/Alle'
```

**Vorteile:** Einfach, deterministisch, kein Extra-Code
**Nachteile:** Bei sehr breiten Fenstern verschwendet die zweite Zeile Platz

### Ansatz B: Dynamisch via `resize`-Event (elegant)

fzf bietet seit 0.36.0:
- `resize`-Event: Feuert bei Fenstergrößenänderung
- `transform-header`-Action: Header dynamisch setzen
- `$FZF_COLUMNS`: Aktuelle Spaltenanzahl

```zsh
--bind 'start,resize:transform-header:
  if (( FZF_COLUMNS < 60 )); then
    printf "Enter: Checkout/Browser | Ctrl+D: Diff\nCtrl+O: Browser | Ctrl+S: Open/Alle"
  else
    printf "Enter: Checkout/Browser | Ctrl+D: Diff | Ctrl+O: Browser | Ctrl+S: Open/Alle"
  fi'
```

**Vorteile:** Optimale Platznutzung bei jeder Fensterbreite
**Nachteile:** Mehr Code pro Funktion, Schwellenwert muss pro Funktion bestimmt werden

### Ansatz C: Helper-Funktion (skalierbar)

Eine zentrale Helper-Funktion, die Header automatisch umbricht:

```zsh
# In fzf.alias oder einer Helper-Datei
fzf-header() {
    local parts=("$@")
    local line="" result=""
    for part in "${parts[@]}"; do
        if (( ${#line} + ${#part} + 3 > FZF_COLUMNS )); then
            result+="${line}\n"
            line="$part"
        else
            [[ -n "$line" ]] && line+=" | "
            line+="$part"
        fi
    done
    result+="$line"
    printf '%s' "$result"
}

# Verwendung:
--bind 'start,resize:transform-header:fzf-header "Enter: Checkout" "Ctrl+D: Diff" "Ctrl+O: Browser" "Ctrl+S: Open/Alle"'
```

**Vorteile:** DRY, einmal implementieren → überall nutzen, automatischer Umbruch
**Nachteile:** Höchste Komplexität, muss bei jedem resize synchron laufen

### Empfehlung

**Ansatz A für Funktionen >55 Zeichen** als Quick Win, **Ansatz C als Ziel** für langfristige Konsistenz. Ansatz B ist der Mittelweg.

Die Schwelle ~55 Zeichen basiert auf: tmux-Split (50%), Standardbreite 120 Spalten → 60 Spalten pro Pane minus Padding/Border.

## Alternativen

1. **Nichts tun** — Akzeptieren, dass Header abgeschnitten werden. User lernen die Keybindings mit der Zeit. ❌ Schlecht für Discoverability.

2. **Header kürzen** — Statt `Enter: Checkout/Browser` nur `⏎:Checkout` etc. Spart Zeichen, reduziert aber Lesbarkeit. ❌ Keine deutsche Beschreibung mehr möglich.

3. **`--header` durch `--border-label` ersetzen** — Label auf dem Border statt separater Header-Zeile. ❌ Noch weniger Platz, keine Mehrzeiligkeit.

4. **`--footer` für Overflow** — Unter die Liste verschieben. ✅ Möglich, aber weniger prominent.

## Zusätzlicher Kontext

- **fzf-Version:** 0.67.0 (Homebrew)
- **Relevante fzf-Features:** `resize`-Event (seit 0.36.0), `transform-header` (seit 0.36.0), `$FZF_COLUMNS` (seit 0.42.0), `start`-Event (seit 0.44.0)
- **fzf Man Page:** `--header=STR` — "The given string will be printed as the sticky header. [...] ANSI color codes are processed even when --ansi is not set."
- **Kein `--header-wrap`:** fzf bietet kein automatisches Header-Wrapping. `--wrap` gilt nur für Listen-Items.
- **`--header-lines`:** Wird im Repo nicht verwendet (0 Vorkommen)
- **Issue #284:** Wenn dort Ctrl+Y zu weiteren Funktionen hinzugefügt wird, werden Header **noch länger** → dieses Issue wird dringlicher

### PR DESCRIPTION
**Pull request by tshofmann**
## Beschreibung

Löst das Problem abgeschnittener fzf-Header in schmalen Terminals (Issue #302). Ein neuer Helper `header-wrap` bricht Header intelligent an semantischen Gruppen um, basierend auf der tatsächlich verfügbaren Breite.

**Kernidee:** `--header` bleibt als Fallback (Initial-Wert), `start,resize:transform-header` ruft `header-wrap` dynamisch auf. Bei Terminal-Resize oder Preview-Toggle wird der Header automatisch neu umbrochen.

### Neuer Helper `terminal/.config/fzf/header-wrap` (308 Zeilen)

- Berechnet verfügbare Breite aus `FZF_COLUMNS` und `FZF_PREVIEW_COLUMNS`
- Preview-Erkennung: seitlich → Separator = 2 Zeichen abziehen; oben/unten → volle Breite
- Greedy-Algorithmus packt semantische Gruppen zeilenweise mit ` | ` als Separator
- Umbruch nur ZWISCHEN Gruppen, nie innerhalb
- 33 Selbsttests (`header-wrap --test`) — Zeilenumbruch, Breitenberechnung, Integration, Edge-Cases
- Bekannte Limitation dokumentiert: `${#var}` zählt Zeichen, nicht Display-Spalten (für Latin/Deutsch korrekt; CJK irrelevant)

### Migration aller 24 fzf-Header

- 10 Alias-Dateien + `init.zsh` migriert
- `rg-live`: Separate `--bind`-Statements (`start:+transform-header` + `resize:transform-header`) — notwendig, weil `start:reload` bereits existiert und `+`-Prefix (append) nicht in komma-separierter Syntax mit `:command` kombinierbar ist (fzf 0.68 Parsing-Limitation, getestet)
- Alle anderen: `start,resize:transform-header` (konsistent)
- `init.zsh` (`FZF_CTRL_R_OPTS`): Korrektes Escaping mit `\"...\"`

### Konsistenz-Normalisierung

- `Esc`-Hinweis aus allen Headers entfernt (fzf-Standard, Platz sparen)
- Toggle-Symbol einheitlich `↔` mit Leerzeichen (nicht `/`)
- Kommentar-Beschreibungen an Header-Text angepasst
- `--bind ` statt `--bind=` normalisiert
- `7zf`: Separator `=` → `:` und Box-Drawing `│` → Pipe `|`
- tldr-Seiten (`fzf.patch.md`, `gh.patch.md`) regeneriert

### CI & Pre-Commit

- Pre-Commit Check 9: header-wrap Selbsttests (Existenz + Ausführbar + 33/33 Tests)
- CI validate.yml: Gleicher Check
- ZSH-Syntax-Check erweitert auf fzf-Helper-Dateien

### Empirische Erkenntnisse (Korrektur zu Issue-Kommentar)

- Separator zwischen Main-Area und Preview = **2 Zeichen** (nicht 6 wie initial vermutet)
- `FZF_COLUMNS` ist Always gleich, unabhängig von Preview-Position
- `FZF_PREVIEW_COLUMNS` ist leer wenn Preview versteckt
- `min_main_width=10` als Schwelle für oben/unten-Erkennung

### Geänderter Umfang

16 Dateien, +419 / -33 Zeilen über 5 Commits:
- `feat: dynamischer fzf-Header-Umbruch (header-wrap)` — Initiale Implementierung
- `test: header-wrap in Testumgebung integrieren` — CI/Pre-Commit
- `fix: Review-Findings beheben (M1, M2, H1, L1, L3, L5)` — 6 von 8 Self-Review-Findings
- `fix: verbleibende Review-Findings beheben (L2, L4)` — Restliche 2 Findings
- `fix: Code- und Doku-Konsistenz (Esc, Toggle-Symbole, Leerzeichen)` — Konsistenz-Audit

## Art der Änderung

- [x] ✨ Neues Feature

## Checkliste

- [x] Ich habe die [Contributing Guidelines](CONTRIBUTING.md) gelesen
- [x] `./.github/scripts/generate-docs.sh --check` ist erfolgreich
- [x] `./.github/scripts/health-check.sh` zeigt keine Fehler
- [x] Neue Aliase/Funktionen haben Beschreibungskommentare
- [x] Bei neuen Tools: Guard-Check vorhanden

## Zusammenhängende Issues

Closes #302

Folge-Issue: #305 (Dokumentations-Drift durch redundante Keybinding-Quellen verhindern)

### header-wrap --test (33/33)

```
header-wrap Selbsttest
══════════════════════

Zeilenumbruch
──────────────
  ✓ Breites Terminal — einzeilig
  ✓ Schmales Terminal — Umbruch
  ✓ Einzelgruppe — immer einzeilig
  ✓ Keine Gruppen — leere Ausgabe
  ✓ gh-pr bei 45 Spalten — Umbruch nach Gruppe 1
  ✓ Exakt passende Breite — einzeilig
  ✓ Ein Zeichen zu wenig — Umbruch
  ✓ Pipe innerhalb Gruppe bleibt erhalten
  ✓ Drei Gruppen — 2+1 Verteilung
  ✓ Sehr schmal — jede Gruppe eigene Zeile
  ✓ Zwei Gruppen exakt passend
  ✓ gh-pr bei voller Breite — einzeilig
  ✓ cmds bei 39 Spalten — Umbruch nach Gruppe 2
  ✓ Einzelgruppe mit Pipe
  ✓ Überbreite Gruppe — wird nicht abgeschnitten
  ✓ available=0 — jede Gruppe eigene Zeile
  ✓ available=1 — jede Gruppe eigene Zeile
  ✓ Leere Gruppe wird übersprungen
  ✓ Nur leere Gruppen — leere Ausgabe
  ✓ Kein Trailing-Newline (Wert)
  ✓ Kein Trailing-Newline (letztes Zeichen)

Breitenberechnung
──────────────────
  ✓ Ohne Preview → volle Breite
  ✓ Preview rechts 50% → Hauptbereich
  ✓ Preview oben 60% → volle Breite
  ✓ Preview versteckt → volle Breite
  ✓ Kein FZF_COLUMNS → Fallback 80
  ✓ Preview rechts 80% → schmaler Hauptbereich
  ✓ 120col Terminal mit Preview rechts 50%
  ✓ Hauptbereich 9 Spalten → seitlich erkannt
  ✓ Hauptbereich 8 Spalten → oben/unten erkannt
  ✓ FZF_PREVIEW_COLUMNS=0 → volle Breite

Integration
───────────
  ✓ Pipeline: 80col+Preview → gh-pr umbricht
  ✓ Pipeline: 80col ohne Preview → gh-pr umbricht (79 > 74)

══════════════════════
Ergebnis: 33/33 bestanden
```

### Pre-Commit (9/9)

```
→ Prüfe Shell-Syntax...
✔ Shell-Syntax OK
→ Prüfe Execute-Berechtigungen...
✔ Execute-Berechtigungen OK
→ Prüfe ob Dokumentation aktuell ist...
✔ Dokumentation ist aktuell
→ Prüfe Alias-Datei-Format...
✔ Alias-Format OK
→ Prüfe Header-Einrückungen...
✔ Header-Einrückungen OK
→ Prüfe Synchronisation platform.zsh ↔ install.sh...
✔ Plattform-Sync OK
→ Prüfe Markdown-Dateien...
✔ Markdown OK
→ Führe Health-Check aus...
✔ Health-Check OK
→ Führe header-wrap Selbsttests aus...
✔ header-wrap Selbsttests OK

✔ Alle Checks bestanden
```

### CODE REVIEW

File: terminal/.config/alias/rg.alias [hunk 1]
```diff
@@ -78,6 +78,8 @@ if command -v fzf >/dev/null 2>&1; then
             --preview "$FZF_HELPER_DIR/preview file {1} {2}" \
             --preview-window 'up,60%,border-bottom,+{2}+3/3,~3' \
             --header='Enter: Datei öffnen | Ctrl+Y: Pfad kopieren' \
+            --bind "start:+transform-header:$FZF_HELPER_DIR/header-wrap 'Enter: Datei öffnen' 'Ctrl+Y: Pfad kopieren'" \
+            --bind "resize:transform-header:$FZF_HELPER_DIR/header-wrap 'Enter: Datei öffnen' 'Ctrl+Y: Pfad kopieren'" \
```
[the remark this answers is not in the document]
Comment (tshofmann) [reply terminal/.config/alias/rg.alias]: Danke für den Vorschlag! Die kombinierte Syntax funktioniert hier leider **nicht** — getestet mit fzf 0.68.0:

```
$ echo "test" | fzf --bind "start:+transform-header,resize:transform-header:echo combined" --filter="test"
→ "unknown action: transform-header" (exit=2)
```

**Ursache:** fzf parst die komma-separierte Bind-Syntax inkorrekt, wenn der `+`-Prefix (append) mit `:command`-Syntax gemischt wird. Das Komma trennt normalerweise `event:action`-Paare, aber bei `start:+transform-header,resize:transform-header:cmd` interpretiert fzf `transform-header` nach dem Komma als eigenständige Action (ohne Command), was fehlschlägt.

**Warum zwei separate `--bind` nötig sind:**
- `start:+transform-header` braucht `+` (append), weil bereits `start:reload` in Zeile 74 existiert — ohne `+` würde das reload überschrieben
- `resize:transform-header` braucht kein `+` (keine bestehende resize-Bindung)
- Da die Events unterschiedliche Prefixe benötigen, müssen sie in separaten `--bind`-Statements stehen

Die zwei Zeilen sind korrekt und beabsichtigt. Alle anderen Funktionen nutzen `start,resize:transform-header` (ohne `+`), weil dort kein bestehendes `start`-Binding existiert.
